### ISSUE TITLE
[VT-42] vit; "x" for execute annotation

### ISSUE THREAD

**Issue by taskwarrior** [scottkosty/vit#64]
_David Patrick on 2013-08-12T16:21:24Z says:_ 

 If one could execute a command or script by selecting an annotation and hitting "x", (skipping any date and/or label) then any number of actions could be "embedded" in any task. 
This is like using taskopen, but more directly executing from vit. 
Attempting this on anything other than an annotation with an executable line would simply cause a "not executable" error.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2013-08-12T16:21:24Z
Modified: 2014-02-09T01:16:51Z
```

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2013-08-13T01:41:42Z says:_ 

 I have not dug in to taskopen yet. Looking at the examples it looks like the argument to taskopen is just the task ID. If so, then you can already do this with shortcuts. For example, to use the 'x' key, put the following in ~/.vitrc:

<pre>
map x=taskopen %
</pre>

or

<pre>
map --wait x=taskopen %
</pre>

If this doesn't cover the common workflow of Taskopen can you provide some examples?

**Comment by taskwarrior** [comment]
_David Patrick on 2013-08-13T02:09:52Z says:_ 

 Those examples look interesting Scott, and I want to learn how that works. That said, I think we may be talking about different things. This specific request (#1336) has almost nothing to do with taskopen, other than it would allow vit to do something taskopen -x option can do, which is "execute annotations". 

This revolves around a rather exotic notion that an annotation could contain a string to be executed, for example

   $ task 142 ann error6: cat ~/.task/messages/6.txt

and then, move the selection to that annotation, under 142, and hitting x executes "cat ~/.task/messages/6.txt"

This actually has some profound handling implications, enabling embedded functionality.

ON THE OTHER HAND, feature #1237 (vit; cmd o [args]) IS about calling taskopen from within vit.

**Comment by taskwarrior** [comment]
_David Patrick on 2013-08-13T02:30:48Z says:_ 

 Scott Kostyshak wrote:
> I have not dug in to taskopen yet. Looking at the examples it looks like the argument to taskopen is just the task ID. If so, then you can already do this with shortcuts. 

Taskopen has evolved into something alarmingly powerful and versatile, although it is focused on link-type annotations. 
It deserves a good look, but it's not obvious what it is does to the casual observer. 

> For example, to use the 'x' key, put the following in ~/.vitrc:
> [...]
> 
> or
> 
> [...]
> 
> If this doesn't cover the common workflow of Taskopen can you provide some examples?

y'know, some variation of that might just work! I'll try it.

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2013-08-14T06:16:44Z says:_ 

 David Patrick wrote:
> Scott Kostyshak wrote:
> > I have not dug in to taskopen yet. Looking at the examples it looks like the argument to taskopen is just the task ID. If so, then you can already do this with shortcuts. 
> 
> Taskopen has evolved into something alarmingly powerful and versatile, although it is focused on link-type annotations. 
> It deserves a good look, but it's not obvious what it is does to the casual observer. 
> 
> > For example, to use the 'x' key, put the following in ~/.vitrc:
> > [...]
> > 
> > or
> > 
> > [...]
> > 
> > If this doesn't cover the common workflow of Taskopen can you provide some examples?
> 
> y'know, some variation of that might just work! I'll try it.

Good, I'm looking forward to hearing your experience. I think the shortcuts are straightforward and can give a lot of flexibility. Did it work?

**Comment by taskwarrior** [comment]
_David Patrick on 2013-09-15T01:58:00Z says:_ 

 Can be done with vit 1.2 and taskopen.

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2013-09-15T06:36:21Z says:_ 

 Fixed at 8d16697d.
